src/frontend/apps/markdown/MarkdownEditor/index.tsx

```diff
@@ -0,0 +1,334 @@
+      markdownDocument,
+      'content',
+      language,
+    );
+    setInitialMarkdownContent(translatedContent);
+    setLocalMarkdownContent(translatedContent);
+    updateEditorContent(translatedContent);
+
+    const translatedRenderedContent = getMarkdownDocumentTranslatedContent(
+      markdownDocument,
+      'rendered_content',
+      language,
+    );
+    setLocalRenderedContent(translatedRenderedContent);
+    setInitialRenderedContent(translatedRenderedContent);
+
+    setLocalIsDraft(markdownDocument.is_draft);
+  }, [markdownDocument, language]);
+
+  // Only change detection here
+  useEffect(() => {
+    if (
+      localTitle !== initialTitle ||
+      localMarkdownContent !== initialMarkdownContent ||
+      localRenderedContent !== initialRenderedContent
+    ) {
+      setContentChanged(true);
+    } else {
+      setContentChanged(false);
+    }
+  }, [
+    localTitle,
+    localMarkdownContent,
+    localRenderedContent,
+    initialTitle,
+    initialMarkdownContent,
+    initialRenderedContent,
+  ]);
+
+  // Detect change to alert user before leaving
+  useEffect(() => {
+    const beforeUnload = (event: BeforeUnloadEvent) => {
+      if (contentChanged) {
+        event.preventDefault();
+        event.returnValue = '';
+      }
+    };
+    window.addEventListener('beforeunload', beforeUnload);
+    return () => window.removeEventListener('beforeunload', beforeUnload);
+  }, [contentChanged]); // We highly rely on any change performed
+
+  // Helpers
+  const canSaveDocument = () =>
+    !!localMarkdownContent && !!localRenderedContent && !!localTitle;
+
+  const saveDocument = () => {
+    if (canSaveDocument()) {
+      saveDocumentTranslations({
+        language_code: language,
+        title: localTitle,
+        content: localMarkdownContent,
+        rendered_content: localRenderedContent!,
+      });
+    }
+  };
+
+  const publishDocument = () => {
+    mutateDocument({
+      is_draft: false,
+    } as MarkdownDocument);
+    // This will invalidate query cache, and refetch all data
+  };
+
+  const updateEditorContent = (content: string) => {
+    if (!codemirrorView) return;
+    const update = codemirrorView.state.update({
+      changes: {
+        from: 0,
+        to: codemirrorView.state.doc.length,
+        insert: content,
+      },
+    });
+    codemirrorView.update([update]);
+    // Warning: this does not clear the state, ie. an "undo"
+    // in the editor will restore the previous content (bad
+    // when we are talking about language change).
+    // `codemirrorView.setState(state)` would be a better solution,
+    // but it would be a bad idea not to make it inside the
+    // `CodeMirrorEditor` component.
+  };
+
+  return (
+    <Box pad="xsmall">
+      <Suspense fallback={<Loader />}>
+        {isSaving ? <Loader /> : null}
+
+        <Box direction="row" height="3rem">
+          <Box direction="row" flex="grow" basis="xxsmall">
+            <TextInput
```

Comment:
add a  `a11ytitle` to this input. Will be allow to find it in the test using `screen.getByRole('textinput', { name: the_a11ytitle_value })`;